provider/image.go

```diff
@@ -370,3 +399,40 @@ func marshalSkipPush(sp resource.PropertyValue) bool {
 	}
 	return sp.BoolValue()
 }
+
+func processLogLine(msg string) (string, error) {
+	var info string
+	var jm jsonmessage.JSONMessage
+	err := json.Unmarshal([]byte(msg), &jm)
+	fmt.Println("HERE IT IS", jm)
+	q.Q(jm)
+	if err != nil {
+		return info, errors.Wrapf(err, "encountered error unmarshalling:")
+	}
+	// process this JSONMessage
+	if jm.Error != nil {
+		if jm.Error.Code == 401 {
+			return info, fmt.Errorf("authentication is required")
+		}
+		return info, errors.Errorf(jm.Error.Message)
+	}
+	if jm.ID != "" {
+		info += jm.ID
+	}
+	if jm.From != "" {
+		info += jm.From
+	}
+	if jm.Progress != nil {
+		info = jm.Status + " " + jm.Progress.String()
+	} else if jm.Stream != "" {
+		info += jm.Stream
+	} else {
+		info += jm.Status
+	}
+	if jm.Aux != nil {
+		infoBytes, _ := json.Marshal(jm.Aux)
+		// because this is unstructured JSON we print out the whole object.
+		info += string(infoBytes) //TODO:
```

Comment:
What's the TODO?

Reply:
unfortunately buildkit dumps all the logs to aux, so we have to process them a bit more.